### ISSUE TITLE
AFNetworkReachabilityManager strange behavior

### ISSUE THREAD

**Issue by AKupetskiy** [AFNetworking/AFNetworking#2827]
Hello! 

I used Reachability manager to track connection to specific domain. I call managerForDomain: to setup manager. After some minutes of debugging i found this strange case in switch (AFNetworkReachabilityManager.m);
```objC
- (void)startMonitoring {
...
   switch (self.networkReachabilityAssociation) {
        case AFNetworkReachabilityForName:
            break;
        case AFNetworkReachabilityForAddress:
        case AFNetworkReachabilityForAddressPair:
...
}
```
Maybe it's just a mistake, or - do we really need that break after first case? With this behavior, as i think, calling 'managerForDomain' is useless. 

Source code of managerForDomain:
```objC
+ (instancetype)managerForDomain:(NSString *)domain {
...
    manager.networkReachabilityAssociation = AFNetworkReachabilityForName;

    return manager;
}
```

Thanks,
Andrew.

**Comment by chadkouse** [comment]
:+1 I get not updates to reachability at all when using host name.

**Comment by kcharwood** [comment]
Here is what I can gather on the history of this issue so far:

* Originally introduced in 3e60343a15bbf02eeabe90d38e0945d38a6acac6 as a result of #1677 and shipped in 2.1.0
* As noted in [this](https://github.com/AFNetworking/AFNetworking/commit/3e60343a15bbf02eeabe90d38e0945d38a6acac6#commitcomment-10204284) comment, @mattt wasn't sure of the original intent of that break statement.
* 99894d4033d9fb8f8904390c860539366c06e7bd was introduced and removed `AFNetworkReachabilityAssociation` entirely. Shipped in 2.5.2
* e92af05 reverted back to the original behavior in 2.5.3

I'm leaving this as a note to myself. Will need to look further into it tomorrow.

**Comment by kcharwood** [comment]
I believe this i fixed in https://github.com/AFNetworking/AFNetworking/pull/3117
